src/Chapter2.hs

```diff
@@ -606,7 +629,10 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate = go
+  where
+    go []     = []
+    go (x:xs) = x : x : go xs
```

Comment:
Wonderful!

Although you can see that the definition of the function now is `duplicate = go`. This means that these two functions are just equal. I think you can directly inline the `go` function into `duplicate`, so you don't have the intermediate helper function 🙂

Reply:
Done !